posts/2025-07-01-25.0.0.7-beta.adoc

```diff
@@ -0,0 +1,227 @@
+```
+
+For more information about MicroProfile Telemetry, see the following links:
+
+- https://github.com/eclipse/microprofile-telemetry[MicroProfile Telemetry]
+- https://github.com/open-telemetry/opentelemetry-specification[specification]
+- https://openliberty.io/docs/latest/docs/latest/microprofile-telemetry.html[documentation]
+    
+    
+// DO NOT MODIFY THIS LINE. </GHA-BLOG-TOPIC> 
+
+// // // // DO NOT MODIFY THIS COMMENT BLOCK <GHA-BLOG-TOPIC> // // // // 
+// Blog issue: https://github.com/OpenLiberty/open-liberty/issues/31811
+// Contact/Reviewer: Azquelt
+// // // // // // // // 
+[#openapi]
+== MicroProfile OpenAPI 4.1
+
+OpenAPI is a standardized way of documenting REST APIs in a JSON or YAML format. MicroProfile OpenAPI helps to generate and serve OpenAPI documentation for REST applications that are built by using JAX-RS or Jakarta restfulWS. This can be useful for developers to test out the API during development, or for people who uses API in production.
+
+In this beta, the new MicroProfile OpenAPI 4.1 is introduced. To use it, first add the feature to your `server.xml`:
+
+```xml
+<featureManager>
+    <feature>mpOpenAPI-4.1</feature>
+</featureManager>
+```
+
+Then, deploy rest application and then see `/openapi` on liberty server to view the OpenAPI documentation, or go to `/openapi/ui` for a user interface that lets to browse through it. To enhance the documentation, use link:https://download.eclipse.org/microprofile/microprofile-open-api-4.1/apidocs/[annotations] in code to add things like textual descriptions to each method. The annotations are available as a maven dependency:
+
+```xml
+<dependency>
+    <groupId>org.eclipse.microprofile.openapi</groupId>
+    <artifactId>microprofile-openapi-api</artifactId>
+    <version>4.1</version>
+    <scope>provided</scope>
+</dependency>
+```
+
+MicroProfile OpenAPI 4.1 is a very small release, which adds:
+
+- The ability to set the link:https://spec.openapis.org/oas/v3.1.0.html#fixed-fields[`jsonSchemaDialect` field on the OpenAPI object] using the link:https://download.eclipse.org/microprofile/microprofile-open-api-4.1/apidocs/org.eclipse.microprofile.openapi/org/eclipse/microprofile/openapi/models/OpenAPI.html#setJsonSchemaDialect(java.lang.String)[model API]
+- New methods `hasExtension` and `getExtension` on link:https://download.eclipse.org/microprofile/microprofile-open-api-4.1/apidocs/org.eclipse.microprofile.openapi/org/eclipse/microprofile/openapi/models/Extensible.html[`Extensible`] in the model API
+
+More information about MicroProfile OpenAPI can be found in
+
+- the liberty link:https://openliberty.io/guides/microprofile-openapi.html[guide]
+- the liberty link:https://openliberty.io/docs/latest/documentation-openapi.html[documentation]
+- the link:https://download.eclipse.org/microprofile/microprofile-open-api-4.1/apidocs/[javadoc]
+- the link:https://download.eclipse.org/microprofile/microprofile-open-api-4.1/microprofile-openapi-spec-4.1.html[specification]
```

Comment:
I think I should have put "Open Liberty" here instead of "liberty"
```suggestion
More information about MicroProfile OpenAPI can be found in

- the Open Liberty link:https://openliberty.io/guides/microprofile-openapi.html[guide]
- the Open Liberty link:https://openliberty.io/docs/latest/documentation-openapi.html[documentation]
- the link:https://download.eclipse.org/microprofile/microprofile-open-api-4.1/apidocs/[javadoc]
- the link:https://download.eclipse.org/microprofile/microprofile-open-api-4.1/microprofile-openapi-spec-4.1.html[specification]
```